source/assocPath.js

```diff
@@ -27,6 +27,8 @@ import isNil from './isNil.js';
  *
  *      // Any missing or non-object keys in path will be overridden
  *      R.assocPath(['a', 'b', 'c'], 42, {a: 5}); //=> {a: {b: {c: 42}}}
+ *      R.assocPath(['a', 1, 0], 42, {a: []}); // => {a: [undefined, [42]]}
+ *      R.assocPath(['a', -1], 42, {a: [1, 2]}); // => {a: [undefined, [1, 42]]}
```

Comment:
No `undefined` in this one:

 `{a: [undefined, [1, 42]]}` -> `{ a: [ 1, 42 ] }`

Reply:
Made a better example